detection_rules/config.py

```diff
@@ -156,10 +157,13 @@ def parse_rules_config(path: Optional[Path] = None) -> RulesConfig:
     if path:
         assert path.exists(), f'rules config file does not exist: {path}'
         loaded = yaml.safe_load(path.read_text())
-    elif CUSTOM_RULES_DIR:
-        path = Path(CUSTOM_RULES_DIR) / '_config.yaml'
-        assert path.exists(), f'_config.yaml file missing in {CUSTOM_RULES_DIR}'
+    elif (path := Path(CUSTOM_RULES_DIR) / '_config.yaml').exists():
         loaded = yaml.safe_load(path.read_text())
+        if 'configuration_details' in loaded:
+            print(loaded['configuration_details'])
+            path.unlink()
+            sys.exit(0)
```

Comment:
Can you explain the purpose of this

Reply:
The only file that actually _requires_ data to not fail on default, is packages.yaml - explicitly the package version, which you could search for.

Additionally, you should not exit from here (especially with 0 code), but raise a specific error, like `ValueError`

Reply:
I'll update the summary.

Reply:
Why do you delete the file? Also please raise an exception vs exiting

Reply:
I would also recommend this:

>The only file that actually requires data to not fail on default, is packages.yaml - explicitly the package version, which you could search for.

over the search for `configuration_details` because if they create it themselves or dont use `--default` this will be missed

Reply:
> Why do you delete the file? Also please raise an exception vs exiting

I mentioned this already in the summary, but essentially, im thinking of an experience where users want defer to use the -use-default option after noticing the default doesn't get them much. With the way the `parse_rules_config` is called in so many different load times in the repo, you'll run into a blocker immediately. Removing the file after trying to use the config generated makes sure that you can call subsequent cli commands. It may be more helpful to run the commands in serial so you can see what I mean. Note: it doesn't remove the file immediately.

Reply:
I don't think we should blindly delete files from users, especially based on assumptions. Better to raise a detailed error (missing data, refer to docs for file) and let them delete it.

Reply:
I was thinking about raising this to the CLI. Which I think I will.

Reply:
So the issue here is that the user doesn't even get a chance to specify by CLI to delete the file because it crashes by assert early in the execution. This time in rule.py

Reply:
I exposed deleting the file to the CLI.